workflow/scripts/heatmap.r

```diff
@@ -0,0 +1,127 @@
+###############################################################################
+#
+#R script to generate a heatmap
+#
+#   AUTHOR: Krish Agarwal
+#   AFFILIATION: University_of_Basel
+#   CONTACT: akrish136@gmail.com
+#   CREATED: 23-07-2020
+#   LICENSE: Apache_2.0
+##############################################################################
+
+# by default: suppress warnings
+options(warn = -1)
+
+# load libraries
+suppressPackageStartupMessages(library("optparse"))
+library(ggplot2)
+
+# list the command-line arguments
+option_list <- list(
+  make_option(c("--input_tsv"),
+    action = "store",
+    dest = "input_tsv",
+    type = "character",
+    help = "location and name of the tsv file"
+  ),
+  make_option(c("--input_sequence"),
+    action = "store",
+    dest = "input_sequence",
+    type = "character",
+    help = "full input_sequence"
+  ),
+  make_option(c("--outfile"),
+    action = "store",
+    dest = "output_tsv",
+    type = "character",
+    help = "location and name of output heatmap"
+  )
+)
+
+# parse command-line arguments
+opt_parser <- OptionParser(
+  usage = "Usage: %prog [OPTIONS] --message [STRING]",
+  option_list = option_list,
+  add_help_option = FALSE,
+  description = ""
+)
+opt <- parse_args(opt_parser)
+
+
+input_sequence <- opt$input_sequence
+input_tsv <- opt$input_tsv
+output_tsv <- opt$output_tsv
+
+x_axis_numbers = 1:nchar(input_sequence)
+
+file = file(input_tsv, "r")
+
+y_axis_labels = c()
+
+while (TRUE) 
+{
+  line = readLines(file, n = 1)
+  if (length(line) == 0) 
+  {
+    break
+  }
+    words = strsplit(line, '\t')
+    pwm_id = words[[1]][1]
+    if(is.element(pwm_id, y_axis_labels) == FALSE && pwm_id != "pwm_id")
+    {
+      y_axis_labels <- c(y_axis_labels, pwm_id)
+    } 
+}
+uniform_data = matrix(as.double("0.00"), length(y_axis_labels), length(x_axis_numbers))
+
+close(file)
+
+file = file(input_tsv, "r")
+
+while (TRUE) 
```

Comment:
Could you comment your code a little? Not every line but maybe every loop? or every code block?